### PR TITLE
small AverageOptimizer enhance.

### PR DESCRIPTION
**Pull request by panyx0718**

### CODE REVIEW

File: python/paddle/fluid/optimizer.py [hunk 1]
```diff
@@ -1110,6 +1110,8 @@ class ModelAverage(Optimizer):
 
     The size of average window is determined by average_window_rate,
     min_average_window, max_average_window and current update times.
+    the effective average window should be larger than min_average_window and
+    larger than min(average_window_rate * num_updates, max_average_window).
```
Comment (wanghaoshuang) [inline python/paddle/fluid/optimizer.py]: 'the' -> 'The' and 'larger'->'less ' ?

Comment (panyx0718) [reply python/paddle/fluid/optimizer.py]: It seems to be larger.
```
    if (num_accumulates >= min_average_window &&
        num_accumulates >= std::min<int64_t>(max_average_window,
                                             num_updates * average_window)) {
```

Comment (panyx0718) [reply python/paddle/fluid/optimizer.py]: I removed my new comments. I suspect the window doesn't control the final effective window. It's triggered by apply. Cannot find a test for it...

Comment (wanghaoshuang) [reply python/paddle/fluid/optimizer.py]:
```
   if (num_accumulates >= min_average_window &&
         num_accumulates >= std::min<int64_t>(max_average_window,
                                              num_updates * average_window)) {
       //  Now the average window is too long, discard the old sum.
       out_sum_3_tensor.device(place) = in_sum_1_tensor + in_sum_2_tensor;
```
This condition is used to check if the average window is too long as the comments in [line 97](https://github.com/PaddlePaddle/Paddle/pull/11761/files#diff-adfe93eee1293eb85675a6a2221d03c1R97)

**Review by wanghaoshuang — APPROVED** [review]
LGTM

**Pull request merged**